src/frontend/src/btc/utils/btc-transactions.utils.ts

```diff
@@ -0,0 +1,29 @@
+import type { BtcTransactionUi } from '$btc/types/btc';
+import type { BtcAddress } from '$lib/types/address';
+import type { BitcoinOutput, BitcoinTransaction } from '$lib/types/blockchain';
+import { nonNullish } from '@dfinity/utils';
+
+export const mapBtcTransaction = ({
+	transaction: { inputs, block_index, out, hash, time },
+	btcAddress
+}: {
+	transaction: BitcoinTransaction;
+	btcAddress: BtcAddress;
+}): BtcTransactionUi => {
+	const isTypeSend = inputs.some(({ prev_out }) => prev_out.addr === btcAddress);
+	const blockIndexAvailable = nonNullish(block_index);
+	const output: BitcoinOutput | undefined = out.find(({ addr }) =>
+		isTypeSend ? addr !== btcAddress : addr === btcAddress
+	);
+
+	return {
+		id: hash,
+		timestamp: time,
+		value: BigInt(output?.value ?? 0),
```

Comment:
Is it correct to default to zero?

Reply:
It's a good question. I think yes, but I can adjust it in case I face something that speaks against it.

Reply:
After double checking the `IcTransaction`, I think should maybe handle the value as potentially `undefined`?

To some extent, I interpret no value displayed differently than zero - i.e. no value displayed might be a bug while zero would lead me to ask myself "WHAT Zero??? Where is my money?" 😉.

Reply:
Ok, agreed 🙂  I have already merged this PR to my feature branch, so I will update it there.